keymaster/ta/parsel.c

```diff
@@ -64,13 +82,16 @@ int TA_deserialize_blob_akms(uint8_t *in, const uint8_t *end,
 			return in - start;
 		}
 		TEE_MemMove(data, in, blob->data_length);
-		in += blob->data_length;
 		blob->data = data;
 	} else {
 		/* Not allocate memory, it can be too large */
 		blob->data = in;
-		in += blob->data_length;
 	}
+	if (ADD_OVERFLOW((uintptr_t)in, blob->data_length, &sum)) {
```

Comment:
We use this directly instead of the overflow functions, since we don't know the end of sum so to say? Correct?

Reply:
Sorry, this change isn't really required as the steps are 1. check for out of bounds 2. add size or len to pointer and/or use pointer as arg in memcpy or TEE_MemMove(). 1 is already done in L71 (or 70) above so we can just add the size here without further checking.


Will remove this part.